components/dashboard/src/components/Modal.tsx

```diff
@@ -41,7 +41,8 @@ export default function Modal(props: {
         return () => {
             window.removeEventListener('keydown', handler);
         };
-    }, []); // Empty array ensures that effect is only run on mount and unmount
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [props.onClose, props.onEnter]);
```

Comment:
Note: This was required because apparently state variables (like `name: string` and `value: string` in `AddVariableModal`) get "captured" in the `onClose` / `onEnter` lambdas, thus causing these functions to always use initial state values (i.e. empty string) instead of updated state values (i.e. whatever name/value a user typed).

The solution was to make this effect depend on `onClose` and `onEnter`, such that, if these functions change (they do change every time a captured state variable changes), the handler gets re-registered properly.